providers/implementations/kem/rsa_kem.c

```diff
@@ -131,16 +131,13 @@ static int rsakem_init(void *vprsactx, void *vrsa,
                        const char *desc)
 {
     PROV_RSA_CTX *prsactx = (PROV_RSA_CTX *)vprsactx;
-    int protect = 0;
 
     if (!ossl_prov_is_running())
         return 0;
 
     if (prsactx == NULL || vrsa == NULL)
         return 0;
 
-    if (!ossl_rsa_key_op_get_protect(vrsa, operation, &protect))
-        return 0;
```

Comment:
Removing this call also removes the check https://github.com/openssl/openssl/blob/0958f5a5bc47e2cca907a3bfaf14059c020324fd/providers/common/securitycheck.c#L41-L50 that `RSAPSAPSS` keys are unsuitable for some operations.

Reply:
Side note:

I find the terminology around RSAVE in the RSA KEM code to be very unfortunate, just because generate and verifyrecover are called *operations* in the NIST document, DOES NOT mean that these are in any sense replacing the actual KEM **encapsulate** and **decapsulate** operations being performed.

Rather this is just a choice of underlying mechanism, a KEM "mode" perhaps, but the documentation around this, and choice of the parameter name to select `RSASVE` just create unnecessary confusion.

It would be good to fix the docs and perhaps even the parameter name to clear this up. And perhaps even make RSASVE be the default mode for the RSA KEM (as the only one supported)...

Cc: @paulidale

Reply:
Argh, this side effect is unfortunate.